lib/person.rb

```diff
@@ -0,0 +1,27 @@
+class Person
+  attr_reader :name,
+              :interests,
+              :supplies
+
+  def initialize(attributes)
+    @name      = attributes[:name]
+    @interests = attributes[:interests]
+    @supplies  = Hash.new {|hash, key| hash[key] = 0}
+  end
+
+  def add_supply(item, value)
+    @supplies[item] += value
+  end
+
+  # def can_build?(craft)
```

Comment:
I would only loop through the craft.supplies_required to check that each supply required is included in this persons @supplies and that the quantity is greater than what is required. Something like:

```
craft.supplies_required.each do |supply, quantity_needed|
  if !@supplies[supply.to_s] || @supplies[supply.to_s] < quantity_needed
    return false
  end
end
true
```